### PR TITLE
general tidy-up

### PR DESCRIPTION
**Pull request by jtnord**
Miscellanous tidyups

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -132,6 +132,16 @@
   </dependencies>
 
   <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <configuration>
+            <reuseForks>false</reuseForks>
```
Comment (jtnord) [inline pom.xml:140]: We are inserting a security provider which is static JVM config - so best to not attempt to reuse forks

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 2]
```diff
@@ -183,15 +184,15 @@ public static PEMEncodable decode(@Nonnull String pem, @Nullable final char[] pa
                     PEMEncryptedKeyPair ekp = (PEMEncryptedKeyPair) object;
                     return new PEMEncodable(kConv.getKeyPair(ekp.decryptKeyPair(dp)));
                 } else {
-                    throw new UnrecoverableKeyException();
+                    throw new UnrecoverableKeyException("Key is passphrase protected, but no passphrase was provided");
```
Comment (jtnord) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java:187]: just makes the error obvious

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 3]
```diff
@@ -183,15 +184,15 @@ public static PEMEncodable decode(@Nonnull String pem, @Nullable final char[] pa
...
                 }
             } else if (object instanceof PKCS8EncryptedPrivateKeyInfo) {
                 if (passphrase != null) {
                     InputDecryptorProvider dp = new JceOpenSSLPKCS8DecryptorProviderBuilder().build(passphrase);
                     PKCS8EncryptedPrivateKeyInfo epk = (PKCS8EncryptedPrivateKeyInfo) object;
                     return new PEMEncodable(kConv.getPrivateKey(epk.decryptPrivateKeyInfo(dp)));
                 } else {
-                    throw new UnrecoverableKeyException();
+                    throw new UnrecoverableKeyException("Key is passphrase protected, but no passphrase was provided");
```
Comment (jtnord) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java:195]: just makes the error obvious

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 4]
```diff
@@ -273,7 +274,7 @@ public static PEMEncodable read(@Nonnull File pemFile) throws IOException, Unrec
     @Nonnull
     public static PEMEncodable read(@Nonnull File pemFile, @Nullable char[] passphrase)
             throws IOException, UnrecoverableKeyException {
-        return decode(FileUtils.readFileToString(pemFile), passphrase);
+        return decode(FileUtils.readFileToString(pemFile, StandardCharsets.UTF_8), passphrase);
```
Comment (jtnord) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java:277]: PEM certs are not platform dependant, also removes a deprecated warning

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 5]
```diff
@@ -284,7 +285,7 @@ public static PEMEncodable read(@Nonnull File pemFile, @Nullable char[] passphra
      * @throws IOException launched if a problem exists generating the PEM information or writing the {@link File}
      */
     public void write(@Nonnull File pemFile) throws IOException {
-        FileUtils.writeStringToFile(pemFile, encode());
+        FileUtils.writeStringToFile(pemFile, encode(), StandardCharsets.UTF_8);
```
Comment (jtnord) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java:288]: PEM certs are not platform dependant, also removes a deprecated warning

File: src/test/java/jenkins/bouncycastle/EncodingDecodingTest.java [hunk 6]
```diff
@@ -54,7 +56,12 @@
 
     @BeforeClass
     public static void setUpBC() {
-        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    @AfterClass
```
Comment (jtnord) [inline src/test/java/jenkins/bouncycastle/EncodingDecodingTest.java:62]: technically not requires as surefire is configured to not reuse forks, but this is good for IDEs

File: src/test/java/jenkins/bouncycastle/FingerPrintTest.java [hunk 7]
```diff
@@ -47,11 +48,17 @@
 
     @BeforeClass
     public static void setUpBC() throws URISyntaxException {
-        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+        Security.addProvider(new BouncyCastleProvider());
         PEM_FILE = new File(
                 EncodingDecodingTest.class.getClassLoader().getResource("private-key-fingerprint.pem").toURI());
     }
 
+    @AfterClass
```
Comment (jtnord) [inline src/test/java/jenkins/bouncycastle/FingerPrintTest.java:56]: technically not requires as surefire is configured to not reuse forks, but this is good for IDEs

File: src/test/java/jenkins/bouncycastle/SecurityProviderInitializerTest.java [hunk 8]
```diff
@@ -25,20 +25,32 @@
...
 import org.jvnet.hudson.test.JenkinsRule;
 
 public class SecurityProviderInitializerTest {
 
+    @BeforeClass
+    public static void validateEnv() {
+        assertNull(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME));
+    }
+
+    @AfterClass
```
Comment (jtnord) [inline src/test/java/jenkins/bouncycastle/SecurityProviderInitializerTest.java:44]: technically not requires as surefire is configured to not reuse forks, but this is good for IDEs

File: src/test/java/jenkins/bouncycastle/SecurityProviderInitializerTest.java [hunk 9]
```diff
@@ -25,20 +25,32 @@
...
+import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 import org.jvnet.hudson.test.JenkinsRule;
 
 public class SecurityProviderInitializerTest {
 
+    @BeforeClass
+    public static void validateEnv() {
+        assertNull(Security.getProvider(BouncyCastleProvider.PROVIDER_NAME));
```
Comment (jtnord) [inline src/test/java/jenkins/bouncycastle/SecurityProviderInitializerTest.java:41]: this was not checking that BC was not alread registered, and so was not actually checking the code was registering BouncyCastle :)

**Review by imonteroperez — APPROVED** [review]
LGTM

**Pull request merged**